tests/components/cloud/test_http_api.py

```diff
@@ -1046,7 +1046,7 @@ async def test_enabling_remote_remote_activation_not_allowed(
     client = await hass_ws_client(hass)
     mock_connect = cloud.remote.connect
     assert not cloud.client.remote_autostart
-    cloud.client.prefs.async_update(remote_allow_remote_enable=False)
+    await cloud.client.prefs.async_update(remote_allow_remote_enable=False)
```

Comment:
If this wasn't awaited before but the test didn't fail, what does the test test? Seems the test isn't doing what it should.

@emontnemery

Reply:
Side note: We have this fixture to help set cloud preferences.

https://github.com/home-assistant/core/blob/b35d7a348bbcb01d1f202e2840558098fc44797c/tests/components/cloud/conftest.py#L164-L178

Reply:
That was my reaction as well, it looks like this is not testing this setting as the following calls enable the connection from the hass instance which would still be allowed, it should be changed to just check the `CloudPrefrences.remote_allow_remote_enable` property or even better call `CloudClient.async_cloud_connect_update(True)` and make sure it raises `RemoteActivationNotAllowed`.

However, this PR on its own is correct as this should be awaited.